app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -94,6 +103,59 @@ public void onClick(View v) {
                         dateTime.get(Calendar.MONTH), dateTime.get(Calendar.DAY_OF_MONTH)).show();
             }
         });
+
+    }
+
+    private void submitMood() throws DescriptionTooLongException {
+        ElasticSearchController.AddMoodTask addMoodTask = new ElasticSearchController.AddMoodTask();
+        Mood mood = new Mood("user"); //TODO: pass logged in user to here
+        if(showPublicOn){
+            mood.setPublic();
+        }
+        else {
+            mood.setPrivate();
+        }
+        if(locationOn) {
+            //TODO: mood.setMapPosition();
+            // this is the setter for latitude and longitude
+        }
+        mood.setEmotionalState(emotionalState);
+        mood.setSocialSit(socialSit);
+        mood.setDescription(String.valueOf(inputMoodDescription.getText()), " -Feeling " + emotionalState); //TODO: We might not need to send this append part to Elasticsearch, but rather add it to our displayed description after we fetch from Elasticsearch
+        mood.setDate(dateTime.getTime());
+        // TODO: mood.setImage();
+        // this is the setter for image
+        addMoodTask.execute(mood);
```

Comment:
User posts are built and sent to Elasticsearch within this function right here. If uploading to Elasticsearch fails, then it is pushed to a local queue. That functionality is outlined more within the ElasticSearchController class